google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java

```diff
@@ -150,28 +147,40 @@ private Subscriber(Builder builder) {
       closeables.add(
           new AutoCloseable() {
             @Override
-            public void close() throws IOException {
+            public void close() {
               executor.shutdown();
             }
           });
     }
-    alarmsExecutor = builder.systemExecutorProvider.getExecutor();
-    if (builder.systemExecutorProvider.shouldAutoClose()) {
+
+    this.numPullers = builder.parallelPullCount;
+    streamingSubscriberConnections = new ArrayList<>(numPullers);
+    TransportChannelProvider channelProvider = builder.channelProvider;
+    if (channelProvider.acceptsPoolSize()) {
+      channelProvider = channelProvider.withPoolSize(numPullers);
+    }
+
+    ExecutorProvider systemExecutorProvider = builder.systemExecutorProvider;
+    if (systemExecutorProvider == null) {
+      systemExecutorProvider =
+          FixedExecutorProvider.create(
+              Executors.newScheduledThreadPool(
+                  Math.max(
+                      SYSTEM_EXECUTOR_MIN_THREAD_COUNT,
```

Comment:
what is the reason for the default min thread of 6? in container environments, a single process may be assigned few cores shares.

Reply:
This is to maintain parity with the current status quo, that the system executor has 6 cores always, by making it have at least 6 cores.  If you have more insight into the 'correct' minimum number of threads, I would appreciate it, but this makes this a noop in the default case where numPullers is 1 for all systems.